### PR TITLE
Move CGC region tags to template file

### PR DESCRIPTION
**Pull request by c2thorn**
This PR removes the logic that inserts region tags and replaces it with region tag detection.
This also modifies an existing example into the one found in https://github.com/terraform-google-modules/terraform-google-network/blob/97b4b9d5fede8aa5b9a62e55162f61570d7ddb91/examples/private_service_connect_google_apis/main.tf

Part of https://github.com/hashicorp/terraform-provider-google/issues/9534

Fixes https://github.com/hashicorp/terraform-provider-google/issues/8317

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by c2thorn** [comment]
@rileykarson I'm not sure why oics change doesn't include the renamed example, it just deleted the old named version.

**Comment by rileykarson** [comment]
Oh, unusual- but the link is still added? Can you check if that logic has drifted? It's unusual the link is present but the backing sample isn't.

File: mmv1/templates/terraform/examples/private_service_connect_google_apis.tf.erb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  region                   = "us-central1"
+  network                  = google_compute_network.network.id
+  private_ip_google_access = true
+}
+# [END vpc_subnet_private_access]
+
+# [START compute_internal_ip_private_access]
+resource "google_compute_global_address" "default" {
+  provider      = google-beta
+  project      = google_compute_network.network.project
```
Comment (rileykarson) [inline mmv1/templates/terraform/examples/private_service_connect_google_apis.tf.erb:23]: super nitpicky, but spacing is off here.

File: mmv1/templates/terraform/resource.html.markdown.erb [hunk 2]
```diff
@@ -100,7 +100,7 @@ state as plain-text. [Read more about sensitive data in state](/docs/state/sensi
      Unless/Until we split our docs, this is a non-issue. -%>
 <% unless object.examples.empty? -%>
   <%- object.examples.reject(&:skip_docs).each do |example| -%>
-    <%- unless example.skip_test -%>
+    <%- unless example.skip_test || !example.test_env_vars.nil? -%>
```
Comment (rileykarson) [inline mmv1/templates/terraform/resource.html.markdown.erb:103]: Should we apply the same `&& e.test_env_vars.any?` as in https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/provider/terraform_oics.rb#L37?

Comment (rileykarson) [reply mmv1/templates/terraform/resource.html.markdown.erb:103]: Not sure if there's any cases where it comes up, but seems prudent to match exactly

Comment (c2thorn) [reply mmv1/templates/terraform/resource.html.markdown.erb]: shouldn't hurt!

**Review by rileykarson — APPROVED** [review]
LGTM. You can also mark fixes for https://github.com/hashicorp/terraform-provider-google/issues/8317 since you rooted that one out (thanks!)

**Pull request merged**